src/Http/ClientEvent.php

```diff
@@ -0,0 +1,87 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http;
+
+use Cake\Http\Client\Response;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Used as a wrapper class to hold request and response data before
+ * the HTTP client actually does the request.
+ */
+class ClientEvent
```

Comment:
This class needs to either extend `Event` or implement the `EventInterface`.

Reply:
There's no error since you are passing this class instance as data for an `Event` instance instead of directly dispatching it :slightly_smiling_face:.

Reply:
Thats why I was asking/was confused with your explenation.
So we literally want a extended Event class with additional functionality just for this kind of event dispatching. Now I understand.

Reply:
I thought when I said `a custom event class` it was clear I meant an actual event class, not a custom class used as DTO :)

Reply:
I can try to get to this on the next weekend. If you instead want to implement this feel free to do it right 😉